src/Traits/IndexTraits/SolrIndexTrait.php

```diff
@@ -61,14 +61,30 @@ public function setService(SolrCoreService $service): self
         return $this;
     }
 
+    /**
+     * Is this Index in debug mode
+     *
+     * @return bool
+     */
+    public function isDebug(): bool
+    {
+        return $this->debug;
+    }
+
     /**
      * Set the debug mode
      *
-     * @param bool $debug
+     * @param bool $debug Set the task in debug mode
+     * @param bool $force Force a task in debug mode, despite e.g. being Live and not CLI
      * @return self
      */
-    public function setDebug(bool $debug): self
+    public function setDebug(bool $debug, bool $force = false): self
     {
+        // Make the debug a configurable, forcing it to always be false from config
+        if (!$force && SolrCoreService::config()->get('debug') === false) {
```

Comment:
`trait CoreServiceTrait` has property `protected $inDebugMode = false;`, I wonder if this is 3rd variable concerning about debug toggling.

Reply:
Yes, it is, good catch. It needs to be unified indeed. Although it has existed as such for a while already, it needs some attention.

Reply:
I am not too concerned about it right now, but absolutely needs to be addressed before a final release.

Reply:
Resolved in #193

Reply:
Haha, I was just reading this comment, and since your avatar is the same style as CodeClimate (black background, white "something"), I was thinking "Since when did CodeClimate become so friendly in the automatic messages?" :rofl:

Reply:
Lol, my response seems robotic, which is good for automation 😲